### ISSUE TITLE
Swc minifier doesn't support the `parse` key in `terserrc`

### ISSUE THREAD

**Issue by aminya** [parcel-bundler/parcel#9041]
# 🐛 bug report

If the `.terserrc` config has the `parse` key, the SWC minifer will error. 

```

@parcel/optimizer-swc: unknown field `parse`, expected one of `compress`, `mangle`, `format`, `ecma`, `keep_classnames`, `keepClassnames`, 
`keep_fnames`, `keepFnames`, `module`, `safari10`, `toplevel`, `sourceMap`, `outputPath`, `inlineSourcesContent`, `emitSourceMapColumns` at line 1 
column 748

  Error: unknown field `parse`, expected one of `compress`, `mangle`, `format`, `ecma`, `keep_classnames`, `keepClassnames`, `keep_fnames`, 
  `keepFnames`, `module`, `safari10`, `toplevel`, `sourceMap`, `outputPath`, `inlineSourcesContent`, `emitSourceMapColumns` at line 1 column 748

```

## 🎛 Configuration (.babelrc, package.json, cli command)

```js
{
  "compress": {
    "global_defs": {
      "process.env.NODE_ENV": "production",
      "process.env.BABEL_ENV": "production",
      "process.env.NODE_DEBUG": false,
      "process.env.RUNNER_DEBUG": "0"
    },
    "ecma": "2018",
    "toplevel": true,
    "hoist_vars": false,
    "hoist_funs": true,
    "pure_getters": true,
    "unsafe": true,
    "unsafe_arrows": true,
    "unsafe_comps": true,
    "unsafe_Function": true,
    "unsafe_math": true,
    "unsafe_symbols": true,
    "unsafe_methods": true,
    "unsafe_proto": true,
    "unsafe_regexp": true,
    "unsafe_undefined": true,
    "passes": 3
  },
  "parse": {
    "ecma": 2020
  },
  "mangle": true,
  "format": {
    "beautify": false,
    "comments": false
  }
}
```

## 🤔 Expected Behavior

The `parse` key should be ignored or handled. 

## 😯 Current Behavior

This errors out. 

## 💁 Possible Solution

Removing the `parse` key.

```
const terserConfig = {...}

delete terserConfig.parse

module.exports = terserConfig
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.9.0
| Node             | 20
| npm/Yarn         |  pnpm 8.5.1
| Operating System | Kubuntu 22.0.4

**Comment by aminya** [comment]
I also realized that some of the other configurations cause issues in the bundled file Such as `unexpected token *`.

This can be reproduced here:
https://github.com/aminya/setup-cpp/tree/cb8940d94559563034c8b8b6c5441983c5f68a0f

**Comment by mischnic** [comment]
Yes this should be fixed (we'll see [who will](https://github.com/swc-project/swc/issues/7455)), but does `parse.ecma` even do anything? That is not documented in Terser's readme

> `unexpected token * `

-> https://github.com/swc-project/swc/issues/7457

**Comment by aminya** [comment]
> does parse.ecma even do anything? That is not documented in Terser's readme

Yeah, it seems to be a legacy option. 
https://github.com/terser/terser/blob/f9567fd5f89201a20ad46c33974efd0a3eb4cee9/lib/parse.js#L1075

**Comment by mischnic** [comment]
This should be fixed now in `@swc/core@1.3.66`, please try that.

**Comment by aminya** [comment]
Thanks! I haven't tested it, but I'll reopen it if there are issues.
